test-data/unit/check-modules.test

```diff
@@ -1943,3 +1943,24 @@ main:2: error: Name 'name' is not defined
 main:3: error: Revealed type is 'Any'
 
 [builtins fixtures/module.pyi]
+
+
+[case testImportCycle]
+from m import One
+reveal_type(One)
+[file m/__init__.py]
+from .one import One
+from .two import Two
+reveal_type(One)
+[file m/one.py]
+class One:
+    pass
+[file m/two.py]
+from m import One
+reveal_type(One)
```

Comment:
Note that references to `One` in annotations still don't work correctly:

```
from m import One
o: One
reveal_type(o)  # Any
```

Similarly, if we make `One` a type alias and use it in an annotation, it will be treated as `Any`.

If we create a subclass of `One` in `m.two`, it seems to have an implicit `Any` base class, which is unexpected.

Reply:
Good catches! I think these are probably fixable by adding more third-pass visits to different node types to propagate type info that we missed in second pass due to the cycle, but it does begin to feel a bit like a game of whack-a-mole.